src/endpoints.rs

```diff
@@ -207,3 +208,63 @@ pub async fn get_region(
             .map(|trainee| trainee.region.clone()),
     }))
 }
+
+#[derive(Serialize)]
+pub struct AttendanceResponse {
+    courses:
+        IndexMap<String, IndexMap<String, IndexMap<String, IndexMap<String, Vec<Attendance>>>>>,
+}
+
+pub async fn fecth_attendance(
+    session: Session,
+    State(server_state): State<ServerState>,
+    OriginalUri(original_uri): OriginalUri,
+) -> Result<Json<AttendanceResponse>, Error> {
+    let all_courses = &server_state.config.courses;
+    let sheets_client = sheets_client(&session, server_state.clone(), original_uri.clone()).await?;
+    let github_org = &server_state.config.github_org;
+    let octocrab = octocrab(&session, &server_state, original_uri).await?;
+    let mut courses: IndexMap<
+        String,
+        IndexMap<String, IndexMap<String, IndexMap<String, Vec<Attendance>>>>,
+    > = IndexMap::new();
+    for (course_name, course_info) in all_courses {
+        let mut course_batches: IndexMap<
+            String,
+            IndexMap<String, IndexMap<String, Vec<Attendance>>>,
+        > = IndexMap::new();
```

Comment:
If you do the type alias thing above, you can rewrite this as:

```suggestion
        let mut course_batches: AttendanceByCohort::new();
```